components/brave_vpn/browser/vpn_service.cc

```diff
@@ -0,0 +1,261 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_vpn/browser/vpn_service.h"
+
+#include "base/json/json_reader.h"
+#include "base/json/json_writer.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/storage_partition.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+
+namespace {
+const char kVpnHost[] = "housekeeping.sudosecuritygroup.com";
+
+const char kAllServerRegions[] = "api/v1/servers/all-server-regions";
+const char kTimezonesForRegions[] = "api/v1.1/servers/timezones-for-regions";
+const char kHostnameForRegion[] = "api/v1/servers/hostnames-for-region";
+const char kCreateSubscriberCredential[] =
+    "api/v1/subscriber-credential/create";
+const char kVerifyPurchaseToken[] = "api/v1.1/verify-purchase-token";
+
+net::NetworkTrafficAnnotationTag GetNetworkTrafficAnnotationTag() {
+  return net::DefineNetworkTrafficAnnotation("vpn_service", R"(
+      semantics {
+        sender: "VPN Service"
+        description:
+          "This service is used to communicate with Guardian VPN apis"
+          "on behalf of the user interacting with the Brave VPN."
+        trigger:
+          "Triggered by user connecting the Brave VPN."
+        data:
+          "Servers, hosts and credentials for Brave VPN"
+        destination: Android
+      }
+    )");
+}
+
+GURL GetURLWithPath(const std::string& host, const std::string& path) {
+  return GURL(std::string(url::kHttpsScheme) + "://" + host).Resolve(path);
+}
+
+std::string CreateJSONRequestBody(const base::Value& dict) {
+  std::string json;
+  base::JSONWriter::Write(dict, &json);
+  return json;
+}
+
+std::string GetSubscriberCredentialFromJson(const std::string& json) {
+  base::JSONReader::ValueWithError value_with_error =
+      base::JSONReader::ReadAndReturnValueWithError(
+          json, base::JSONParserOptions::JSON_PARSE_RFC);
+  base::Optional<base::Value>& records_v = value_with_error.value;
+  const base::Value* subscriber_credential =
+      records_v->FindKey("subscriber-credential");
+  return subscriber_credential->GetString();
+}
+
+}  // namespace
+
+VpnService::VpnService(content::BrowserContext* context)
```

Comment:
How about making factory passing `SharedURLLoaderFactory` instead of `context`?
With it, we can remove `content` dependency also.